File: done/FindPivotIndex.java
package done;

import java.util.Arrays;

public class FindPivotIndex {
    public int pivotIndex(int[] nums) {
        int left = 0;
        int right = Arrays.stream(nums).sum();
        for (int k = 0; k < nums.length; k++) {
            right -= nums[k];
            if (left == right) {
                return k;
            }
            left += nums[k];
        }

        return -1;
    }
}

File: done/Find Pivot Index.kt
fun main() {
    println(pivotIndex(intArrayOf(1,7,3,6,5,6)))
}

fun pivotIndex(nums: IntArray): Int {
    var i = 0; var left = 0; var right = nums.sum()
    while (i < nums.size) {
        right -= nums[i]
        if (left == right) return i
        left += nums[i]
        i++
    }
    return -1
}